src/main/java/bio/terra/cli/context/Workspace.java

```diff
@@ -0,0 +1,241 @@
+package bio.terra.cli.context;
+
+import bio.terra.cli.command.exception.UserActionableException;
+import bio.terra.cli.context.utils.Printer;
+import bio.terra.cli.service.WorkspaceManagerService;
+import bio.terra.workspace.model.WorkspaceDescription;
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * POJO class that represents a workspace. This class is serialized to disk as part of the global
+ * context. It is also used as the user-facing JSON output for commands that return a workspace.
+ */
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+public class Workspace {
+  private static final Logger logger = LoggerFactory.getLogger(Workspace.class);
+
+  // properties of the workspace
+  public final UUID id;
+  public final String name; // not unique
+  public final String description;
+  public final String googleProjectId;
+
+  // name of the server where this workspace exists
+  public final String serverName;
+
+  // email of the user that loaded the workspace to this machine
+  public final String terraUserEmail;
+
+  // list of resources (controlled & referenced)
+  private List<Resource> resources;
+
+  @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
+  private Workspace(
+      @JsonProperty("id") UUID id,
+      @JsonProperty("name") String name,
+      @JsonProperty("description") String description,
+      @JsonProperty("googleProjectId") String googleProjectId,
+      @JsonProperty("serverName") String serverName,
+      @JsonProperty("terraUserEmail") String terraUserEmail,
+      @JsonProperty("resources") List<Resource> resources) {
+    this.id = id;
+    this.name = name;
+    this.description = description;
+    this.googleProjectId = googleProjectId;
+    this.serverName = serverName;
+    this.terraUserEmail = terraUserEmail;
+    this.resources = resources;
+  }
+
+  /**
+   * Create a new workspace and set it as the current workspace.
+   *
+   * @param name optional display name
+   * @param description optional description
+   */
+  public static Workspace create(String name, String description) {
+    // call WSM to create the workspace object and backing Google context
+    WorkspaceDescription createdWorkspace =
+        new WorkspaceManagerService().createWorkspace(name, description);
+    logger.info("Created workspace: {}", createdWorkspace);
+
+    // convert the WSM object to a CLI object
+    Workspace workspace = fromWSMObject(createdWorkspace);
```

Comment:
Ditto consider returning the `Workspace` from the `WorkspaceManagerService` instead of a `WorkspaceDescription`.

Reply:
I decided to keep the `WorkspaceManagerService` as a wrapper around the client library, and do deserialization in the internal object classes (e.g. `Workspace`). This keeps the wrapper class pretty thin (just an enhanced client library) and puts de/serialization in the internal object classes, which is a little more consistent with how it now does de/serialization to the format for command input/output and the format for writing to disk.